Automattic/wp-calypso
# Add `reformat-files:css` script

Adds `reformat-files:css` script that runs `stylelint --fix` on files.

There's this note [in the docs](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/cli.md#autofixing-errors):

>It's an experimental feature. It currently does not respect special comments for disabling stylelint within sources (e. g. `/* stylelint-disable */`). Autofixing will be applied regardless of these comments.
>
>If you're using both these special comments and autofixing, please run stylelint twice as a temporary solution. On the first run, some violations could be missed, or some violations might be reported incorrectly.

Solution might be not run this together with `reformat-files` command and let folks run this only manually.

#### Changes proposed in this Pull Request

* Add new `reformat-files:css` script which always exists with `0` — otherwise if there are errors left, `--fix` [will exit with `2`](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/cli.md#exit-codes) and make it seem like `reformat-files` failed. `-c` in `reformat-files` is meant to let `reformat-files:js` finish even if `:css` would exit with non-0 earlier, so I'm fine also changing it to throw an error at the end.
* Add `.json` to config file so that it'll get run through Prettier ([docs about `.json` support](https://stylelint.io/user-guide/configuration/#loading-the-configuration-object))
* Run some files through it.

#### Testing instructions
Run:
- `npm run reformat-files`
  - Observe `reformat-files:js` finish even if `:css` finishes earlier
  - Linter throws some errors/warnings about stuff it cannot autofix but doesn't fail the NPM script with error. 
  - A bunch of files have changed (happy to include these in the PR, but I made a separate: https://github.com/Automattic/wp-calypso/pull/32386)
- `npm run reformat-files:css`
